include/swift/IDE/Formatting.h

```diff
@@ -0,0 +1,107 @@
+//===--- Formatting.h -----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_FORMATCONTEXT_H
+#define SWIFT_FORMATCONTEXT_H
+
+#include "swift/AST/AST.h"
+#include "swift/AST/ASTWalker.h"
+#include "swift/Frontend/Frontend.h"
+#include "swift/Basic/SourceManager.h"
+#include "swift/IDE/SourceEntityWalker.h"
+#include "swift/Subsystems.h"
+
+namespace swift {
+namespace ide {
+
+struct CodeFormatOptions {
+  bool UseTabs = false;
+  unsigned IndentWidth = 4;
+  unsigned TabWidth = 4;
+};
+
+/// \brief Returns the offset (in bytes) to the start of \p LineIndex
+size_t getOffsetOfLine(unsigned LineIndex, StringRef Text);
+
+/// \brief Returns the offset to the first Non-WhiteSpace Character
+size_t getOffsetOfTrimmedLine(unsigned LineIndex, StringRef Text);
+
+/// \brief Returns the Text on \p LineIndex, excluding Leading WS
+StringRef getTrimmedTextForLine(unsigned LineIndex, StringRef Text);
+
+/// \brief Returns the number of spaces at the begining of \p LineIndex
+/// or if indenting is done by Tabs, the number of Tabs * TabWidth
+size_t getExpandedIndentForLine(unsigned LineIndex, CodeFormatOptions Options,
+                                StringRef Text);
+
+class LineRange {
```

Comment:
This seems to have been renamed in the same commit as moving the code into place.  It would be great if it happened earlier so that this commit was really just moving code around.  If there are other modifications happening here, I think the same applies unless the changes are somehow tied to the act of moving the code.  Does that make sense?